### PR TITLE
Add slurm queue test availability

### PR DESCRIPTION
**Pull request by victorusu**
These tests expect
* all partitions listed to be present in a given system (even if
they are not accessible)
* at least one node per partition to be in allocated, reserved, and/or
idle state
* no more than 10% of compute nodes assigned to a partition to be in
a state other than allocated, reserved, and/or idle

### CODE REVIEW

**Review by ekouts — CHANGES_REQUESTED** [review]
lgtm in general, but maybe we can add the dependencies that you mentioned in slack. If this test fails there is no point on running the `HostnameCheck` since it will block the ci

**Comment by vkarak** [comment]
> - at least one node per partition to be in allocated, reserved, and/or
idle state
> - no more than 10% of compute nodes assigned to a partition to be in
a state other than allocated, reserved, and/or idle

I don't understand the difference of these two. It seems that the second one overrides the first one.

**Review by vkarak — CHANGES_REQUESTED** [review]

File: checks/system/slurm/slurm.py [hunk 1]
```diff
@@ -335,32 +334,34 @@ class SlurmQueueStatusCheck(rfm.RunOnlyRegressionTest):
...
     def assert_partition_exists(self):
-        num_matches = sn.count(sn.findall(fr'^{self.partname}.*', self.stdout))
+        num_matches = sn.count(
+                sn.findall(fr'^{re.escape(self.slurm_partition)}.*',
+                self.stdout))
         return sn.assert_gt(num_matches, 0,
-                            msg=f'{self.partname} not defined for partition '
-                                f'{self.current_system.name}')
+                            msg=f'{self.slurm_partition!r} not defined for '
+                                f'partition {self.current_system.name}')
```
Comment (vkarak) [inline checks/system/slurm/slurm.py:346]: Perhaps?

```suggestion
                                f'partition {self.current_partition.fullname}')
```

**Review by vkarak — COMMENTED** [review]
Lgtm, I have just a minor comment.

**Comment by victorusu** [comment]
> lgtm in general, but maybe we can add the dependencies that you mentioned in slack. If this test fails there is no point on running the `HostnameCheck` since it will block the ci

@ekouts, to make this happen, I need to change the `HostnameCheck` fundamentally... This I will do in another PR.

**Review by ekouts — APPROVED** [review]

**Comment by victorusu** [comment]
@jenkins-cscs retry all

**Review by teojgo — APPROVED** [review]

**Pull request merged**